conductor-client/src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt

```diff
@@ -113,9 +114,18 @@ class DataDeletionService(
 
     @Timed
     override fun authCheckForEntitySetsAndNeighbors(entitySetIds: Set<UUID>, deleteType: DeleteType, principals: Set<Principal>, entityKeyIds: Set<UUID>?) {
-        val srcDstEntitySets = entitySetIds.filter {
-            !entitySetManager.getEntitySet(it)!!.flags.contains(EntitySetFlag.ASSOCIATION)
-        }.toSet()
+
+        val srcDstEntitySets = mutableSetOf<UUID>()
+        entitySetIds.forEach { esid ->
+            entitySetManager.getEntitySet(esid)?.let {
+                if (!it.flags.contains(EntitySetFlag.ASSOCIATION)) {
+                    srcDstEntitySets.add(esid)
+                }
+            } ?: run {
+                throw IllegalArgumentException("Unable to perform delete on $entitySetIds because $esid does not exist")
+            }
```

Comment:
It's not guaranteed that this function will be called only during a delete.

Reply:
What do you mean by that? The function accepts a delete type as a parameter, and checks whether caller has correct permissions on affected entity sets and their property types based on the delete type.